src/f32/funcs.rs

```diff
@@ -85,6 +92,28 @@ pub(crate) mod sse2 {
     // _ps_const_ty!(PS_COSCOF_P2, f32x4, 4.166_664_6e-2);
     // _ps_const_ty!(PS_CEPHES_FOPI, f32x4, 1.273_239_5); // 4 / M_PI
 
+    _ps_const_ty!(g_XMNegativeZero, u32x4, 0x80000000);
+    _ps_const_ty!(g_XMPi, f32x4, std::f32::consts::PI);
+    _ps_const_ty!(g_XMHalfPi, f32x4, std::f32::consts::FRAC_PI_2);
+    _ps_const_ty!(g_XMSinCoefficients0 , f32x4, -0.16666667, 0.0083333310, -0.00019840874, 2.7525562e-06);
+    _ps_const_ty!(g_XMSinCoefficients1 , f32x4, -2.3889859e-08, -0.16665852 /*Est1*/, 0.0083139502 /*Est2*/, -0.00018524670 /*Est3*/);
+    _ps_const_ty!(g_XMOne, f32x4, 1.0);
+    _ps_const_ty!(g_XMTwoPi, f32x4, std::f32::consts::PI * 2.0);
+    _ps_const_ty!(g_XMReciprocalTwoPi , f32x4, 0.159154943);
+
+    macro_rules! XM_PERMUTE_PS {
+        ($v:expr, $c:expr) => {
+            _mm_shuffle_ps($v, $v, $c)
+        }
+    }
+
+    // XMVectorNegativeMultiplySubtract
+    macro_rules! XM_FMADD_PS {
+        ($a:expr, $b:expr, $c:expr) => {
+            _mm_fmadd_ps($a, $b, $c)
```

Comment:
Need to check the fma target feature is available to use this.

Reply:
I've updated these detect target CPU features.

Reply:
In addition to the request to make this an inline function and rename it, there are `mul_add` methods on `Vec3` and `Vec4` that could call this when `vec3sse2` and `vec4sse2` are enabled respectively.